tests/test_hamiltonian_sparse.py

```diff
@@ -0,0 +1,120 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for the expval method of the :mod:`pennylane_lightning_kokkos.LightningKOKKOS` device.
+"""
+import math
+
+import numpy as np
+import pennylane as qml
+import pytest
+from pennylane import DeviceError
+from pennylane_lightning_kokkos import LightningKokkos
+
+
+class TestHamiltonianExpval:
+    def test_hamiltionan_expectation(self, qubit_device_3_wires, tol):
+
+        dev = LightningKokkos(wires=3, c_dtype=np.complex128)
```

Comment:
Using dev = qml.device("lightning.kokkos", wires=3, c_dtype=np.complex128) may fix  the kokkos::finalize() issue.